lib/portlayer/storage/vsphere/volume.go

```diff
@@ -51,11 +51,17 @@ func NewVolumeStore(op trace.Operation, storeName string, s *session.Session, ds
 		return nil, err
 	}
 
-	dm, err := disk.NewDiskManager(op, s, DetachAll)
+	dm, err := disk.NewDiskManager(op, s)
 	if err != nil {
 		return nil, err
 	}
 
+	if DetachAll {
```

Comment:
I think DetachAll should be during pl.storage initialization rather than in the NewX methods - this will cause havoc if we end up trying to create scratch at the same time as a NewVolumeStore for example - ditto if we ever have dynamic config.